### PR TITLE
NEO-2327 // Table: update wording for select/clear functionality

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Table story](https://deploy-preview-479--neo-react-library-storybook.netlify.app/?path=/story/components-table--editable-data)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] ~tagged `@avaya-dux/dux-design` if any visual changes have occurred~
- [x] tagged `@avaya-dux/dux-devs`

Matt requested updated wording and functionality for the select and clear options on the Table. Specifically, some updated wording and only showing the available options instead of showing everything and disabling the unavailable ones. [See the ticket](https://avaya.atlassian.net/browse/NEO-2327) for more details.

This is the dev PR. I will create another PR for design after this. 

`@avaya-dux/dux-devs`: refactor and expand the functionality of select/clear Table buttons

- Yang: removed JS code that did hover functionality (removed `const [checkBoxOver, setCheckBoxOver] = useState(false); `, and updated `src/components/Table/Table_shim.css`). It acts the same as [what is currently out](https://neo-react-library-storybook.netlify.app/?path=/story/components-table--editable-data) there, just with pure CSS now thanks to `:has`!
- This is a large PR because I did a major refactor of `TableHeader.tsx` to make it manageable. The short version is that we have six options available that show when appropriate.
  - SelectCurrentPage
  - ClearCurrentPage
  - SelectAllPages
  - ClearAllPages
  - SelectAll
  - ClearAll
- I updated the tests, but the updates are effectively just for the new phrasing Matt requested. 

## Summary by CodeRabbit

- **New Features**
  - Added options for selecting items per page in the table component (5, 10, or 50 items).
  
- **Bug Fixes**
  - Improved test cases for pagination and data refresh behavior, ensuring accurate messages upon data removal and selection states.

- **Refactor**
  - Streamlined the selection logic in the `ClearSelectionRow` component for better performance.
  - Restructured the `TableHeader` component for improved readability and modularity.

- **Style**
  - Enhanced CSS styles for the table header to improve hover effects and user interaction.

- **Documentation**
  - Updated localization strings for clarity and improved user experience in the table component.

### CODE REVIEW

File: src/components/Table/Table_shim.css [hunk 1]
```diff
@@ -89,7 +89,8 @@
 	font-size: 16px;
 }
 
-.neo-table th.neo-table__dnd-th__hover {
+/* set background color on dnd-th when checkbox-th is hovered */
+.neo-table th.neo-table__dnd-th:has(~ .neo-table-checkbox-th:hover) {
```
Comment (yangAtSpoken) [inline src/components/Table/Table_shim.css:93]: nice, but ~ means any sibling, how about +, which is the very next sibling?

File: src/components/Table/TableComponents/TableBody.tsx [hunk 2]
```diff
@@ -100,70 +99,35 @@ const ClearSelectionRow: TableBodyComponentType = ({
...
+								handleRowToggled,
+							)
 						}
 					>
-						{translations.selectAll}
-					</ClearButton>
+						{allTableEnabledRowsAreSelected
+							? translations.clearAll
+							: translations.selectAll}
+					</ClearButton>{" "}
```
Comment (yangAtSpoken) [inline src/components/Table/TableComponents/TableBody.tsx:129]: remove {" "}

**Review by yangAtSpoken — APPROVED** [review]
Love the refactoring to consolidate the menu construction logic.

File: src/components/Table/TableComponents/TableBody.tsx (hunk 2 above)
Comment (joe-s-avaya) [reply src/components/Table/TableComponents/TableBody.tsx:129]: Can't remove this as it is needed to add space between the button and the following text:

File: src/components/Table/Table_shim.css (hunk 1 above)
Comment (joe-s-avaya) [reply src/components/Table/Table_shim.css]: 🤷 we're targeting a specific class and action (`.neo-table-checkbox-th:hover`), so it really doesn't matter, but it doesn't hurt to be more specific. Updated.

**Pull request merged**